api/src/pcapi/core/finance/api.py

```diff
@@ -2137,22 +2199,21 @@ def generate_and_store_invoice_legacy(
     ).one()
 
     with log_elapsed(logger, "Generated invoice HTML", log_extra):
-        invoice_html = _generate_invoice_html_legacy(invoice, batch)
-    with log_elapsed(logger, "Generated and stored PDF invoice", log_extra):
+        if is_debit_note:
+            invoice_html = _generate_debit_note_html(invoice, batch)
```

Comment:
[bug ?] Ça ne devrait pas plutôt être :

```suggestion
            invoice_html = _generate_debit_note_html_legacy(invoice, batch)
```

Reply:
Merci pour ce commentaire. Du coup, ça m'a permis de détecter un autre oublie dans `generate_and_store_invoice` : 
```python
    with log_elapsed(logger, "Generated invoice HTML", log_extra):
        if is_debit_note:
            invoice_html = _generate_debit_note_html(invoice, batch)
        else:
            invoice_html = _generate_invoice_html(invoice, batch)
            ```